landlock/restrict.go

```diff
@@ -9,23 +9,38 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+// downgrade calculates the actual ruleset to be enforced given the
+// current kernel's Landlock ABI level.
+//
+// It establishes that opt.accessFS ⊆ handledAccessFS ⊆ abi.supportedAccessFS.
 func downgrade(handledAccessFS AccessFSSet, opts []PathOpt, abi abiInfo) (AccessFSSet, []PathOpt) {
 	handledAccessFS = handledAccessFS.intersect(abi.supportedAccessFS)
 
 	resOpts := make([]PathOpt, len(opts))
 	copy(resOpts, opts)
 	for i := 0; i < len(resOpts); i++ {
+		// In case that "refer" is requested on a path, we
```

Comment:
@l0kod downgrade() is the part that implements "best effort" mode - with this change I am making sure that if the user asks for 'refer', the reparenting of files is always possible, even if the kernel does not support V2. -- if the user asks for 'refer' and the kernel just has V1, it'll have to fall back to doing nothing, so that reparenting works.